lib/IRGen/ClassLayout.h

```diff
@@ -96,14 +136,23 @@ class ClassLayout {
 
   bool isFixedLayout() const { return IsFixedLayout; }
 
-  bool isFixedSize() const { return IsFixedSize; }
+  bool isFixedSize() const {
+    return !(Options.contains(ClassMetadataFlags::ClassHasMissingMembers) ||
+             Options.contains(ClassMetadataFlags::ClassHasResilientMembers) ||
+             Options.contains(ClassMetadataFlags::ClassHasGenericLayout) ||
+             Options.contains(ClassMetadataFlags::ClassHasObjCAncestry));
+  }
 
   bool doesMetadataRequireInitialization() const {
-    return MetadataRequiresInitialization;
+    return (Options.contains(ClassMetadataFlags::ClassHasResilientAncestry) ||
+            Options.contains(ClassMetadataFlags::ClassHasGenericAncestry) ||
+            Options.contains(ClassMetadataFlags::ClassHasResilientMembers) ||
+            Options.contains(ClassMetadataFlags::ClassHasMissingMembers));
```

Comment:
`ClassHasMissingMembers` doesn't really make sense here; that's only something that can happen if you're talking about someone *else's* class. But maybe *assuming* the metadata *might* require initialization is worth it.

Reply:
`ClassHasGenericLayout` doesn't require initialization?

Reply:
We sometimes compute a ClassLayout for other classes too, like when doing direct stored property access. So missing members can come up here.

The predicates are meant to be mutually exclusive, so I need to add am assert to rule out the resilient ancestry and generic ancestry case here.